src/repositories/ImmutableRepository.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  BaseEntity,
+  DeepPartial,
+  EntityManager,
+  EntityTarget,
+  FindOneOptions,
+  FindOptionsRelations,
+  FindOptionsWhere,
+} from 'typeorm';
+import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity.js';
+
+import { AbstractRepository } from './AbstractRepository';
+import {
+  EntryNotFoundAfterInsertException,
+  IllegalArgumentException,
+  InsertionException,
+} from './errors';
+
+type CreateBody<T extends BaseEntity> = DeepPartial<T>;
+
+/**
+ * Abstract class representing a repository for immutable entities.
+ * Extends the AbstractRepository class to provide additional functionality for adding and retrieving entities.
+ *
+ * @template T The type of the entity.
+ */
+export abstract class ImmutableRepository<T extends BaseEntity> extends AbstractRepository<T> {
+  /** The primary key of the entity used during the find. */
+  protected readonly primaryKey: keyof T;
+
+  /**
+   * @param primaryKey The primary key of the entity, used during the findOne.
+   * @param entity The concrete entity the repository will manage.
+   * @param manager The entity manager used to handle transactions.
+   */
+  constructor(primaryKey: keyof T, entity: EntityTarget<T>, manager?: EntityManager) {
+    super(entity, manager);
+    this.primaryKey = primaryKey;
+  }
+
+  /********************************** Public Interfaces **********************************/
+  /**
+   * Adds a new entity to the repository using the provided body.
+   *
+   * @param body The body containing the entity data.
+   * @returns A promise that resolves to the newly created entity.
+   */
+  public abstract addOne(body: CreateBody<T>): Promise<T>;
+
+  /**
+   * Retrieves an entity with the specified primary key.
+   *
+   * @param primaryKey The primary key of the entity to retrieve.
+   * @returns A promise that resolves to the entity if found, or null if not found.
+   */
+  public async getOne(primaryKey: string): Promise<T | null> {
+    return await this.findOne(primaryKey);
+  }
+
+  /********************************** Subclass Methods **********************************/
+  /**
+   * Throws an IllegalArgumentException if the given primary key is undefined or an empty array.
+   *
+   * @param primaryKey The primary key to check.
+   */
+  protected throwsIfPKIsUndefined(primaryKey: string | string[]) {
```

Comment:
I'm not sure if the name of this function is enough accurate since it also throw if the PK array is empty.
What if a repository want to override this function so it can check the string respect an email format ?

What do you think about this one ? 
```suggestion
  protected throwsIfPKIsInvalid(primaryKey: string | string[]) {
```